File: frontend/src/types/auditLog.ts
export interface AuditLogItem {
  id: string
  target_id: string
  action: string
  old_value: Record<string, unknown>
  new_value: Record<string, unknown>
  reason: string
  ip_address: string
  user_agent: string
  created_at: string
}

export interface ProcedureAuditLogItem extends AuditLogItem {
  procedure_group_id: string | null
}

export interface AuditLogPage {
  total: number
  page: number
  page_size: number
  items: AuditLogItem[]
}

export interface ProcedureAuditLogPage {
  total: number
  page: number
  page_size: number
  items: ProcedureAuditLogItem[]
}

export interface AuditLogFilter {
  target_id?: string
  action?: string
  date_from?: string
  date_to?: string
  ip_address?: string
  page: number
  page_size: number
}

File: backend/app/schemas/audit_log.py
"""审计日志响应 schema（api-specification §5.9）。"""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class AuditLogItem(BaseModel):
    """单条文件夹审计日志。"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    target_id: str
    action: str
    old_value: dict[str, Any] = Field(default_factory=dict)
    new_value: dict[str, Any] = Field(default_factory=dict)
    reason: str = ""
    ip_address: str = ""
    user_agent: str = ""
    created_at: datetime


class ProcedureAuditLogItem(AuditLogItem):
    """单条程序审计日志（额外含 procedure_group_id）。"""

    procedure_group_id: str | None


class AuditLogPage(BaseModel):
    """文件夹审计日志分页响应。"""

    total: int
    page: int
    page_size: int
    items: list[AuditLogItem]


class ProcedureAuditLogPage(BaseModel):
    """程序审计日志分页响应。"""

    total: int
    page: int
    page_size: int
    items: list[ProcedureAuditLogItem]
